### PR TITLE
Add get_optimizer and set_optimizer to BenchmarkModel API

### PR DESCRIPTION
**Pull request by janeyx99**
non ghstack version of #1452

### CODE REVIEW

**Comment by chuanqi129** [comment]
Glad to see that we have add this feature, it will give us a chance to optimize the `optimizer` of train mode. May I know the landing plan of this feature? @janeyx99 @xuzhao9

**Comment by janeyx99** [comment]
currently working on it! aiming on landing it by the end of this week or start of next

**Comment by chuanqi129** [comment]
> currently working on it! aiming on landing it by the end of this week or start of next

Thanks for the update. I have a small suggestion, could we also add the `get_optimizer()` and `set_optimizer()` into https://github.com/pytorch/benchmark/blob/main/torchbenchmark/models/ADDING_MODELS.md#a-minimal-new-model-addition to guide how to add a `train` mode model into benchmark?

File: torchbenchmark/models/ADDING_MODELS.md [hunk 1]
```diff
@@ -37,9 +37,16 @@ A bare miminum example you can follow is https://github.com/pytorch/benchmark/tr
 The functions you specifically need to implement are 
 1. `__init__()` which is responsible for initalizing your `nn.Module`
 2. `get_module()` which is responsible for returning the initialized `nn.Module` and an example input
-3. `train()` which is a training loop, you can return a `NotImplementedError()` if your example is inference only
+3. `train()` which is a training loop, you can return a `NotImplementedError()` if your example is inference only. If your
+   training loop can be encapsulated by a `forward()`, `backward()`, and `optimizer_step()`, you need not redefine `train()`.
+   Instead, please make sure your model provides functions `forward()`, `backward()`, and `optimizer_step()` along with an
+   attribute `self.optimizer` which will be chained together for testing, see `invoke_staged_train_test()` for details.
```
Comment (janeyx99) [inline torchbenchmark/models/ADDING_MODELS.md:43]: @chuanqi129 Any thoughts on these instructions?

Comment (chuanqi129) [reply torchbenchmark/models/ADDING_MODELS.md:43]: Great, thanks for the update
